src/main/perl/FileEditor.pm

```diff
@@ -58,13 +66,45 @@ sub _is_valid_source
     return -f $fn;
 }
 
+sub _is_reference_newer
+{
+    my ($self) = @_;
+    my $is_newer = 0;   # Assume false
+    if (  exists(*$self->{options}->{source}) && $self->_is_valid_source(*$self->{options}->{source}) ) {
```

Comment:
How to easily distinguish a pipe from a standard file? If we agree that supporting pipes as `source` make sense, I propose that if it is a pipe, it is always considered newer than the existing file.